src/services/refactors/extractType.ts

```diff
@@ -249,4 +251,23 @@ namespace ts.refactor {
         changes.insertNodeBefore(file, firstStatement, factory.createJSDocComment(/* comment */ undefined, factory.createNodeArray(concatenate<JSDocTag>(templates, [node]))), /* blankLineBetween */ true);
         changes.replaceNode(file, selection, factory.createTypeReferenceNode(name, typeParameters.map(id => factory.createTypeReferenceNode(id.name, /* typeArguments */ undefined))));
     }
+
+    function removeCommentsFromTypeNode(node: TypeNode) {
+        let members = factory.createNodeArray<Node>();
+
+        switch (node.kind) {
+            case SyntaxKind.UnionType:
+                members = (node as UnionTypeNode).types;
+                break;
+
+            case SyntaxKind.TypeLiteral:
+                members = (node as TypeLiteralNode).members;
+                break;
+        }

```

Comment:
I think this fix is too limited - we really have to remove the comments deeply from a type.

Reply:
I make changes to support removing comments from node's descendants, I am not sure which other types need to be removed or is there a better way to do it?